cairo_programs/cairo/hints/test_less_than_or_equal_true.cairo

```diff
@@ -0,0 +1,5 @@
+fn main() -> bool {
+    let a = 10_u32;
+    let b = 12_u32;
+    a <= b && a <= a
+}
```

Comment:
Rename this file to `span_slice.cairo`

The Cairo function that uses the `TestLessThanOrEqual` hint is the `array_slice`, when slicing a span.

How to find such functions to test a hint though ?
- [Search](https://github.com/search?q=repo%3Astarkware-libs%2Fcairo+%22hint+TestLessThanOrEqual%22&type=code) for a libfunc in the Cairo compiler that uses that hint: "hint TestLessThanOrEqual"
- Find one that is not using too many hints, or unimplemented ones, here [`build_array_slice`](https://github.com/starkware-libs/cairo/blob/86b6b559ea15359ffaf35b3aa6474836179c57ce/crates/cairo-lang-sierra-to-casm/src/invocations/array.rs#L311)
- [Search ](https://github.com/search?q=repo%3Astarkware-libs%2Fcairo+%22array_slice%22&type=code)for the corresponding function name in the Cairo corelib `array_slice()`
- Check if the function is part of the public API or find a function that uses it, here [`slice` for Span](https://github.com/starkware-libs/cairo/blob/86b6b559ea15359ffaf35b3aa6474836179c57ce/corelib/src/array.cairo#L363-L374)
- Make a Cairo program that uses this function

Usually, the Cairo corelib gives usage example to use this function.

A Cairo program like this can be enough:

```cairo
fn main() {
    let span = array![1, 2, 3].span();
    let _slice = span.slice(0, 1);
}
```

Reply:
Done, thank you for the detailed explanation :)